sdk/python/kfp_tekton/compiler/main.py

```diff
@@ -82,6 +82,7 @@ def compile_pyfile(pyfile, function_name, output_path, type_check, tekton_pipeli
     _compile_pipeline_function(pipeline_funcs, function_name, output_path, type_check, tekton_pipeline_conf)
   finally:
     del sys.path[0]
+    sys.modules.pop(os.path.splitext(filename)[0])
```

Comment:
Since Python caches previsously loaded modules, this change was required to allow using the `compile_pyfile()` method multiple times within the same Python process and enable loading different pipeline scripts from files with the same ("module") name, i.e. we have multiple `condition.py` samples.

Without this change, we would get these errors:
```
ValueError trying to compile /Users/ckadner/PycharmProjects/kfp-tekton_ckadner/samples/flip-coin/condition.py: A function with @dsl.pipeline decorator is required in the py file.
ValueError trying to compile /Users/ckadner/PycharmProjects/kfp-tekton_ckadner/samples/flip-coin-custom-task/condition.py: A function with @dsl.pipeline decorator is required in the py file.
```